tests/test_base36.nim

```diff
@@ -0,0 +1,121 @@
+import unittest2
+import ../stew/base36
+
+when defined(nimHasUsed): {.used.}
+
+func hexToBytes*(a: string, result: var openArray[byte]) =
+  doAssert(len(a) == 2 * len(result))
+  var i = 0
+  var k = 0
+  var r = 0
+  if len(a) > 0:
+    while i < len(a):
+      let c = a[i]
+      if i != 0 and i %% 2 == 0:
+        result[k] = r.byte
+        r = 0
+        inc(k)
+      else:
+        r = r shl 4
+      case c
+      of 'a'..'f':
+        r = r or (10 + ord(c) - ord('a'))
+      of 'A'..'F':
+        r = r or (10 + ord(c) - ord('A'))
+      of '0'..'9':
+        r = r or (ord(c) - ord('0'))
+      else:
+        doAssert(false)
+      inc(i)
+    result[k] = r.byte
+
+func fromHex*(a: string): seq[byte] =
+  doAssert(len(a) %% 2 == 0)
```

Comment:
`%%` is specifically unsigned modulo, is there a reason to use that instead of `mod` which isn't quasi-type-coercive? `len` returns a signed `int`, not an unsigned `int`.